### PR TITLE
Reworking the style to get a more coherent look

### PR DESCRIPTION
**Pull request by Euregan**

### CODE REVIEW

**Comment by Euregan** [comment]
#### 📝Changed routes:


| [**/app**](https://oeaki-git-reworking-style.euregan.now.sh/app) | [**/**](https://oeaki-git-reworking-style.euregan.now.sh/) |
|:-:|:-:|
|<a href="https://oeaki-git-reworking-style.euregan.now.sh/app">Screenshot of /app</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/rPYeRc07UgHPxv_IutJj3_hZW44_.png">(view full size)</a> |<a href="https://oeaki-git-reworking-style.euregan.now.sh/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/KmIXnQDvnijbJts4C8d1J0NKuwc_.png">(view full size)</a> |



| [**/lobby/\[id\]**](https://oeaki-git-reworking-style.euregan.now.sh/lobby/[id]) |
|:-:|
|<a href="https://oeaki-git-reworking-style.euregan.now.sh/lobby/[id]">Screenshot of /lobby/[id]</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/9XAkHq2qGDYARX52S7hudHEkkVQ_.png">(view full size)</a> |







Commit 2a4de71d6d5bf851b31a4475b811e262f7f6988b (https://oeaki-git-reworking-style.euregan.now.sh).

File: components/UI.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    <ChakraInput
+        borderColor="var(--border-color)"
+        borderRadius="var(--active-border-radius)"
+        boxSizing="border-box"
+        value={value}
+        onChange={onChange}
+        type={type}
+        id={id}
+        name={name}
+        placeholder={placeholder}
```
Comment (HALLERPierre) [inline components/UI.js:66]: You can use rest props

File: components/UI.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                align-self: stretch;
+            }
+        `}</style>
+    </div>
+);
+
+export const Columns = ({ children, className }) => (
+    <div className={'columns ' + (className || '')}>
+        {children}
+        <style jsx global>{`
```
Comment (HALLERPierre) [inline components/UI.js:28]: It does not need to be global

File: components/UI.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import { Input as ChakraInput } from '@chakra-ui/core';
+
+export const Rows = ({ children, className }) => (
+    <div className={'rows ' + (className || '')}>
```
Comment (HALLERPierre) [inline components/UI.js:4]: Why not use String template ?

Comment (Euregan) [reply components/UI.js:4]: No reason in particular :D

**Comment by Euregan** [comment]
> About the jsx style,you can use emotion instead of `style` approach. What is your advise on it?

We could use emotion, be what would be the point? JSX CSS fits our needs perfectly: We write perfectly valid CSS, using time proven techniques, and Webpack just bundles it while making sure it's scoped.
The problem with Emotion is the same problem with Chakra's approach: You have to write your component's style in the middle of the DOM.
React has always been good at separating logic from the DOM representation of the view, which is why it doesn't define any JSX way of looping, or conditionals: it all happens in JS. Functional components took this one way further: now most components preprocess data at the top of the function/component, while the bottom is simply declaring the HTML resulting from said processing. This gives the reader clarity when reading a function, knowing that everything is split quite well.
Now, enters Emotion, and Chakra: They put CSS in the bloody middle of the HTML. Quite simply, now, instead of passing props to components, you also pass them style, which means there is no clear separation between the logic flowing between components, and the styling. The view is back to being a mix of style and structure, and the reader can't understand what's what at a glance, he has to actively read every prop to be sure nothing is left out.
JSX style however is usually put at the bottom of the component, which means structure and style are well split, and happens in a specific `<style>` tag, allowing for obvious, at a glance, understanding of what's what.

So, my opinion is, just do CSS in a specific `<style>` tag :D

**Pull request merged**